src/matrand_model.jl

```diff
@@ -0,0 +1,82 @@
+export mat_rand_model
+
+using Noise
+
+function mat_rand(m::Int, n::Int, r::Int, sr::Float64, va::Float64, vb::Float64, c::Float64)
+  xl = rand(Uniform(-0.1, 0.3), m, r)
+  xr = rand(Uniform(-0.1, 0.3), n, r)
+  xs = xl * xr'
+  Ω = findall(<(sr), rand(m, n))
+  B = xs[Ω]
+  B = (1 - c) * add_gauss(B, va, 0; clip = true) + c * add_gauss(B, vb, 0; clip = true)
+  return xs, B, Ω
+end
+
+function matrix_completion_model(xs, B, Ω)
+  m, n = size(B)
+  res = zeros(eltype(B), m, n)
```

Comment:
Keep res as a vector
```suggestion
  res = zeros(eltype(B), m * n)
```